backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.Match;
+import com.google.sps.data.Participant;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+* Servlet that returns some example content.
+*/
+@WebServlet("/search-match")
+public class SearchMatchServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get participant ldap
+    UserService userService = UserServiceFactory.getUserService();
+    String email = userService.getCurrentUser().getEmail();
+    String ldap = email.split("@")[0];
+
+    // Create and sort queries by time
+    Query query = new Query("Match").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Convert list of entities to list of matches
+    List<Match> matches = new ArrayList<Match>();
```

Comment:
What do you think about creating a separate Match object for the purposes of sending/returning data (to the front-end or for notification purposes) or maybe limiting the fields this one holds to reduce overhead since each functionality requires different data to be returned

Reply:
I'm not sure I understand what you mean. Could you please clarify?

Reply:
Yes. For the notifications part, we need to provide it with the two usernames/ldaps of the people who are matched and for the frontend we need to provide it a match status (true/false on whether or not it is a match), so I was thinking that instead of returning the entire match object, we could instead return just the two usernames or a true/false so we are not sending back extra data that we are not planning to use. (I think we could make it even simpler by sending back the two matched usernames in either case as well).

Reply:
Got it, now it returns:
- matchStatus: false if it doesn't exists
- matchStatus: true, firstParticipantLdap: __, secondParticipantLdap:__, duration: __ if it does